### ISSUE TITLE
Icons for custom Kubeflow components in pipeline editor missing

### ISSUE THREAD

**Issue by lehrig** [elyra-ai/elyra#2723]
**Describe the issue**
Icons for custom Kubeflow components missing in Kubeflow pipeline editor.

**To Reproduce**
1. Add custom components to Elyra
2. Open the Kubeflow pipeline editor and note the missing icons:
![Bildschirmfoto 2022-05-12 um 10 33 11](https://user-images.githubusercontent.com/9555494/168028222-f45f5d6a-58c9-4404-82e6-5397dbcfa7f4.png)

**Expected behavior**
Kubeflow icons should appear on the left-hand side of the component palette for custom components.

**Deployment information**
Describe what you've deployed and how:
 - Elyra version: 3.8.0
 - Operating system: linux (RHEL 8.4 on ppc64le; SELinux enabled)
 - Installation source: custom container image (https://github.com/lehrig/kubeflow-ppc64le-notebook-images -> quay.io/ibm/kubeflow-notebook-image-ppc64le:elyra3.8.0-py3.8-tensorflow-cpu2.7.0)
 - Deployment type: Kubeflow (v1.5.0) Notebook Server on OpenShift v4.8

**Comment by ptitzler** [comment]
@lehrig could you please check the browser's console log if there are any error messages that would provide a clue? I've checked 3.8.0 and the latest dev build and everything shows up as expectedin my env.

@ajbozarth any idea what might be going on?

**Comment by lehrig** [comment]
@ptitzler sure, here's my console log.

<details><summary>Log Output</summary>
<pre>
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/metadata/runtime-images
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/local
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/runtimes/types
12321.53a58cbe4a1811710e52.js?v=53a58cbe4a1811710e52:1 LSP: is_visible failed TypeError: Cannot read properties of null (reading 'content')
    at de.context_from_active_document (321.53a58cbe4a1811710e52.js?v=53a58cbe4a1811710e52:1:54862)
    at ue.get_context (321.53a58cbe4a1811710e52.js?v=53a58cbe4a1811710e52:1:58001)
    at ue.is_visible (321.53a58cbe4a1811710e52.js?v=53a58cbe4a1811710e52:1:58228)
    at isVisible (321.53a58cbe4a1811710e52.js?v=53a58cbe4a1811710e52:1:56503)
    at e.isVisible (jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2:1350197)
    at e.get [as isVisible] (jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2:1621742)
    at l (jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2:1618196)
    at Object.o [as search] (jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2:1617898)
    at t.onUpdateRequest (jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2:1611697)
    at t.e.processMessage (jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2:1570040)
is_visible @ 321.53a58cbe4a1811710e52.js?v=53a58cbe4a1811710e52:1
isVisible @ 321.53a58cbe4a1811710e52.js?v=53a58cbe4a1811710e52:1
e.isVisible @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
get @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
l @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
o @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
t.onUpdateRequest @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.processMessage @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
b @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
t @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
y @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
requestAnimationFrame (async)
_ @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
n @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.update @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
m @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
l @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.emit @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
l @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
d @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
factory @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e._execute @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
a @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
requestAnimationFrame (async)
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
l @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
s @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
Promise.then (async)
l @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
d @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.schedule @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
t.invoke @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
_onLayoutModified @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
_onCurrentChanged @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
m @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
l @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.emit @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e._setWidgets @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e._evtFocus @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.handleEvent @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
onActivateRequest @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.processMessage @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
b @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
t @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
y @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
requestAnimationFrame (async)
_ @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
n @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.fit @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
attachWidget @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
t.insertWidget @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
insertItem @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
insertItem @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
ge @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
je @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
y @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
createNewWidget @ lib_index_js.f934d0c4d02d3a96949b.js:1122
createNew @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
createWidget @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
_createOrOpenDocument @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
open @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
openOrReveal @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
Promise.then (async)
execute @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
e.execute @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
(anonymous) @ lib_index_js.f934d0c4d02d3a96949b.js:2266
(anonymous) @ lib_index_js.f934d0c4d02d3a96949b.js:2072
__awaiter @ lib_index_js.f934d0c4d02d3a96949b.js:2068
(anonymous) @ lib_index_js.f934d0c4d02d3a96949b.js:2240
Promise.then (async)
execute @ lib_index_js.f934d0c4d02d3a96949b.js:2240
e.execute @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
p @ jlab_core.ace489612104403d9b5e.js?v=ace489612104403d9b5e:2
qe @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
Je @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
en @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
Yr @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
Xr @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
(anonymous) @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
Ae @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
We @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
tl @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
ft @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
ct @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
n.unstable_runWithPriority @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
ha @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
Fe @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
it @ 3935.4159b022aa6d82e44127.js?v=4159b022aa6d82e44127:2
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/runtimes/types
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/local/notebook/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/local/python-script/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/local/r-script/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/notebook/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/python-script/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/r-script/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/url-catalog:cec4378bda86/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/url-catalog:1af3f72374e8/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/url-catalog:84e0aaeb351a/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/url-catalog:6d1f66a58ba2/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/url-catalog:a7f4ae260b2d/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/url-catalog:578e920e2d9a/properties
services_lib_index_js.aaf13bfe5d4db411d632.js:485 Sending a GET request to https://kubeflow.apps.edb-p1306.cecc.ihost.com/notebook/user-example-com/my-ide/elyra/pipeline/components/KUBEFLOW_PIPELINES/url-catalog:6ef1a032a17f/properties
</pre>
</details>

**Comment by ajbozarth** [comment]
@lehrig Could you also share what the application log (in the terminal) shows when you get that error in the console?

**Comment by ptitzler** [comment]
I ran the container image using the following command in my local environment (it is understood that some results might differ, depending on how/where the image is run):

```
docker run -it -p 8888:8888 quay.io/ibm/kubeflow-notebook-image-ppc64le:elyra3.8.0-py3.8-tensorflow-cpu2.7.0      
```

Two observations:
 - the image has Elyra 3.7.0 installed according to `elyra-pipeline --version`, which I guess was not your intention, given the image tag
   
   ![image](https://user-images.githubusercontent.com/13068832/168147392-d8f6e995-cfb8-4552-8cdb-e804d6b97319.png)

 - the icons are rendered as expected (I launched the image multiple times to see if the behavior was intermittend)
   
   ![image](https://user-images.githubusercontent.com/13068832/168147665-a73ec256-1e1c-4281-a007-4a1552ba0228.png)

**Comment by lehrig** [comment]
- Elyra version: you're right, I'm gonna check that
- It looks like the image itself is correct but that the overall setup is hindering images to load. We are using OpenShift and configured everything via Red Hat Service Mesh (Istio). It could be that some `NetworkPolicy` or `AuthorizationPolicy` is hindering the Jupyter pod to fetch images. Would that theoretically be possible?

**Comment by ajbozarth** [comment]
I've opened https://github.com/elyra-ai/elyra/pull/2728

It seems we weren't using the base_url config if it was set when accessing icons we host on the server

**Comment by lehrig** [comment]
Thanks @ajbozarth - I guess I'll verify in the next Elyra release.

(Elyra version: found a bug in our builds where Elyra is downgraded when installing Tensorflow...going to rebuild our image stack)

**Comment by lehrig** [comment]
<img width="256" alt="Bildschirmfoto 2022-06-08 um 21 03 48" src="https://user-images.githubusercontent.com/9555494/172696362-65cd1b97-2eeb-4e30-a40f-213167f45578.png">

...just confirming it's fixed with Elyra 3.9.0. Thanks!
